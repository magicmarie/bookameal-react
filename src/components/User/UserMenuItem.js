import React from "react";

// menu card: customer side
const UserMenuItem = props => (
  <div className="menu-item">
    <div className="row menuitem">
      <div className="col-sm-6">
        <p>{props.mealName}</p>
      </div>
      <div className="col-sm-3">
        <p>
          <span className="badge badge-secondary">{props.price}</span>
        </p>
      </div>
      <div className="col-sm-3">
        <p>
          <button
            to="#"
            className="btn add"
            title="add to cart"
            onClick={() =>
              props.handleAddCart(
                props.mealId,
                props.mealName,
                props.price,
                props.adminName,
                props.id
              )
            }
          >
            Add to Cart
          </button>
        </p>
      </div>
    </div>
  </div>
);
export default UserMenuItem;
