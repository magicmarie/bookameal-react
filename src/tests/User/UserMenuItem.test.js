import React from "react";
import { shallow } from "enzyme";
import UserMenuItem from "../../components/User/UserMenuItem";

describe("UserCart component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<UserMenuItem handleAddCart={jest.fn()} />);
  });
  it("renders successfully", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders successfully", () => {
    const handleEvent = jest.fn();
    wrapper.find("button.add").simulate("click", handleEvent);
  });
});
