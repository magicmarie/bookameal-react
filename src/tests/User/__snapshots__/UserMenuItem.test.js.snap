// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCart component renders successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserMenuItem
    handleAddCart={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row menuitem"
>
        <div
                className="col-sm-6"
        >
                <p />
        </div>
        <div
                className="col-sm-3"
        >
                <p>
                        <span
                                className="badge badge-secondary"
                        />
                </p>
        </div>
        <div
                className="col-sm-3"
        >
                <p>
                        <button
                                className="btn add"
                                onClick={[Function]}
                                title="add to cart"
                                to="#"
                        >
                                Add to Cart
                        </button>
                </p>
        </div>
</div>,
      "className": "menu-item",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-sm-6"
>
            <p />
</div>,
          <div
            className="col-sm-3"
>
            <p>
                        <span
                                    className="badge badge-secondary"
                        />
            </p>
</div>,
          <div
            className="col-sm-3"
>
            <p>
                        <button
                                    className="btn add"
                                    onClick={[Function]}
                                    title="add to cart"
                                    to="#"
                        >
                                    Add to Cart
                        </button>
            </p>
</div>,
        ],
        "className": "row menuitem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p />,
            "className": "col-sm-6",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              <span
                            className="badge badge-secondary"
              />
</p>,
            "className": "col-sm-3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="badge badge-secondary"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "badge badge-secondary",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              <button
                            className="btn add"
                            onClick={[Function]}
                            title="add to cart"
                            to="#"
              >
                            Add to Cart
              </button>
</p>,
            "className": "col-sm-3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn add"
                onClick={[Function]}
                title="add to cart"
                to="#"
>
                Add to Cart
</button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add to Cart",
                "className": "btn add",
                "onClick": [Function],
                "title": "add to cart",
                "to": "#",
              },
              "ref": null,
              "rendered": "Add to Cart",
              "type": "button",
            },
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row menuitem"
>
          <div
                    className="col-sm-6"
          >
                    <p />
          </div>
          <div
                    className="col-sm-3"
          >
                    <p>
                              <span
                                        className="badge badge-secondary"
                              />
                    </p>
          </div>
          <div
                    className="col-sm-3"
          >
                    <p>
                              <button
                                        className="btn add"
                                        onClick={[Function]}
                                        title="add to cart"
                                        to="#"
                              >
                                        Add to Cart
                              </button>
                    </p>
          </div>
</div>,
        "className": "menu-item",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-sm-6"
>
              <p />
</div>,
            <div
              className="col-sm-3"
>
              <p>
                            <span
                                          className="badge badge-secondary"
                            />
              </p>
</div>,
            <div
              className="col-sm-3"
>
              <p>
                            <button
                                          className="btn add"
                                          onClick={[Function]}
                                          title="add to cart"
                                          to="#"
                            >
                                          Add to Cart
                            </button>
              </p>
</div>,
          ],
          "className": "row menuitem",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p />,
              "className": "col-sm-6",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                <span
                                className="badge badge-secondary"
                />
</p>,
              "className": "col-sm-3",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="badge badge-secondary"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "badge badge-secondary",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                <button
                                className="btn add"
                                onClick={[Function]}
                                title="add to cart"
                                to="#"
                >
                                Add to Cart
                </button>
</p>,
              "className": "col-sm-3",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn add"
                  onClick={[Function]}
                  title="add to cart"
                  to="#"
>
                  Add to Cart
</button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add to Cart",
                  "className": "btn add",
                  "onClick": [Function],
                  "title": "add to cart",
                  "to": "#",
                },
                "ref": null,
                "rendered": "Add to Cart",
                "type": "button",
              },
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
